soh/soh/Enhancements/tts/tts.cpp

```diff
@@ -0,0 +1,724 @@
+                snprintf(arg, sizeof(arg), "%d", gSaveContext.rupees);
+                auto translation = GetParameritizedText("rupees", TEXT_BANK_KALEIDO, arg);
+                SpeechSynthesizer::Instance->Speak(translation.c_str(), GetLanguageCode());
+            } else if (CHECK_BTN_ALL(input->press.button, BTN_DRIGHT)) {
+                //TODO: announce timer?
+            }
+        }
+        
+        uint16_t cursorIndex = (pauseCtx->pageIndex == PAUSE_MAP && !inDungeonScene) ? PAUSE_WORLD_MAP : pauseCtx->pageIndex;
+        if (prevCursorIndex == cursorIndex &&
+            prevCursorSpecialPos == pauseCtx->cursorSpecialPos &&
+            prevCursorPoint[cursorIndex] == pauseCtx->cursorPoint[cursorIndex]) {
+            return;
+        }
+        
+        prevCursorSpecialPos = pauseCtx->cursorSpecialPos;
+        
+        if (pauseCtx->cursorSpecialPos > 0) {
+            return;
+        }
+        
+        switch (pauseCtx->pageIndex) {
+            case PAUSE_ITEM:
+            {
+                char arg[8]; // at least big enough where no s8 string will overflow
+                switch (pauseCtx->cursorItem[PAUSE_ITEM]) {
+                    case ITEM_STICK:
+                    case ITEM_NUT:
+                    case ITEM_BOMB:
+                    case ITEM_BOMBCHU:
+                    case ITEM_SLINGSHOT:
+                    case ITEM_BOW:
+                        snprintf(arg, sizeof(arg), "%d", AMMO(pauseCtx->cursorItem[PAUSE_ITEM]));
+                        break;
+                    case ITEM_BEAN:
+                        snprintf(arg, sizeof(arg), "%d", 0);
+                        break;
+                    default:
+                        arg[0] = '\0';
+                }
+                
+                if (pauseCtx->cursorItem[PAUSE_ITEM] == 999) {
+                    return;
+                }
+                
+                std::string key = std::to_string(pauseCtx->cursorItem[PAUSE_ITEM]);
+                auto translation = GetParameritizedText(key, TEXT_BANK_KALEIDO, arg);
+                SpeechSynthesizer::Instance->Speak(translation.c_str(), GetLanguageCode());
+                break;
+            }
+            case PAUSE_MAP:
+                if (inDungeonScene) {
+                    if (pauseCtx->cursorItem[PAUSE_MAP] != PAUSE_ITEM_NONE) {
+                        std::string key = std::to_string(pauseCtx->cursorItem[PAUSE_MAP]);
+                        auto translation = GetParameritizedText(key, TEXT_BANK_KALEIDO, nullptr);
+                        SpeechSynthesizer::Instance->Speak(translation.c_str(), GetLanguageCode());
+                    }
+                } else {
+                    std::string key = std::to_string(0x0100 + pauseCtx->cursorPoint[PAUSE_WORLD_MAP]);
+                    auto translation = GetParameritizedText(key, TEXT_BANK_KALEIDO, nullptr);
+                    SpeechSynthesizer::Instance->Speak(translation.c_str(), GetLanguageCode());
```

Comment:
can we just have `Speak` take in the string and do `c_str()` stuff deeper in instead?